src/feature-feedback.ts

```diff
@@ -0,0 +1,608 @@
+  }
+
+  private resetState() {
+    this.vote = undefined;
+    this.voteSubmitted = false;
+    this.error = undefined;
+    this.voteNeedsChoosing = false;
+    this.comments.value = '';
+  }
+
+  private async showPopup() {
+    if (this.voteSubmitted) return;
+
+    this.resetState();
+    this.setupResizeObserver();
+    this.setupScrollObserver();
+    this.setupEscapeListener();
+    this.positionPopup();
+    this.isOpen = true;
+    await this.setupRecaptcha();
+  }
+
+  private closePopup() {
+    this.disconnectResizeObserver();
+    this.stopScrollObserver();
+    this.removeEscapeListener();
+    this.isOpen = false;
+  }
+
+  private positionPopup() {
+    const betaRect = this.betaButton.getBoundingClientRect();
+    const popupRect = this.popup.getBoundingClientRect();
+    const windowWidth = window.innerWidth;
+    const windowHeight = window.innerHeight;
+    const windowCenterX = windowWidth / 2;
+    const windowCenterY = windowHeight / 2;
+    if (betaRect.left < windowCenterX) {
+      this.popupTopX = betaRect.right - 20;
+    } else {
+      this.popupTopX = betaRect.left + 20 - popupRect.width;
+    }
+    this.popupTopX = Math.max(0, this.popupTopX);
+    if (this.popupTopX + popupRect.width > windowWidth) {
+      this.popupTopX = windowWidth - popupRect.width;
+    }
+
+    if (betaRect.top < windowCenterY) {
+      this.popupTopY = betaRect.bottom - 10;
+    } else {
+      this.popupTopY = betaRect.top + 10 - popupRect.height;
+    }
+  }
+
+  private setupEscapeListener() {
+    if (!this.boundEscapeListener) return;
+    document.addEventListener('keyup', this.boundEscapeListener);
+  }
+
+  private removeEscapeListener() {
+    if (!this.boundEscapeListener) return;
+    document.removeEventListener('keyup', this.boundEscapeListener);
+  }
+
+  private handleEscape(e: KeyboardEvent) {
+    if (e.key === 'Escape') {
+      this.closePopup();
+    }
+  }
+
+  private stopScrollObserver() {
+    document.removeEventListener('scroll', this.boundScrollListener!);
+  }
+
+  private get popupTemplate() {
+    return html`
+      <div
+        id="popup-background"
+        class=${this.isOpen ? 'open' : 'closed'}
+        @click=${this.backgroundClicked}
+        @keyup=${this.backgroundClicked}
+      >
+        <div
+          id="popup"
+          style="left: ${this.popupTopX}px; top: ${this.popupTopY}px"
+        >
+          <form @submit=${this.submit} id="form" ?disabled=${this.processing}>
+            <div id="prompt">
+              <div id="prompt-text">${this.prompt}</div>
+              <button
+                @click=${(e: Event) => {
+                  e.preventDefault();
+                  this.vote = this.vote === 'up' ? undefined : 'up';
+                }}
+                ?disabled=${this.processing}
+                class="vote-button upvote-button ${this
+                  .upvoteButtonClass} ${this.voteNeedsChoosing ? 'error' : ''}"
+                tabindex="0"
```

Comment:
TIL: safari won't focus buttons even with an explicitly set `tab index` attribute
<img width="526" alt="image" src="https://user-images.githubusercontent.com/7840857/159634634-4016e936-2928-4184-8436-b882b8a99f50.png">

and here I thought that buttons have a default of `tabindex=0` 😄